### DISCUSSION TITLE
ru_gen.py simple_analysis() parameters

### DISCUSSION THREAD

**Discussion by jennieli421** [LooseLab/readfish#255] [General]
In ru_gen.py, `simple_analysis` function, there are these two parameters
```
  throttle : int or float
        The number of seconds interval between requests to the ReadUntilClient
    unblock_duration : int or float
        Time, in seconds, to apply unblock voltage
```
I wonder how they affect the enrichment process and may I able to modify them.

**Comment by mattloose** [comment]
You can modify these but we don't advise it.

The "throttle" setting does not need not be less than the current setting as you risk overwhelming the system. You can test these on playback and see what the effects are.

The "unblock_duration" is something that you can only test on a live run and specifies how long the reverse voltage is applied for. Changing this from the default is likely to impact flowcell performance and so we don't advise tweaking it as you can't test it on playback.

**Comment by alexomics** [comment]
Both parameters are modifiable using the options `--throttle` and `--unblock-duration` when you call `readfish targets ...`.

By default the throttle is 0.4 seconds and the unblock duration is 0.1 seconds. 

Increasing the unblock duration will result in applying the unblock voltage for longer, this _may_ have an adverse effect on the pores as the channel will be unblocking for longer than necessarily needed. In an ideal scenario readfish will not be unblocking molecules that have sequenced more than ~2kb.

The throttle is there to allow tuning with the `break_reads_after_seconds` setting in the MinKNOW sequencing config. The lower this number the more requests that readfish will make.

**Comment by jennieli421** [comment]
For `unblock_duration`, if it is decreased, will the pores eject reads faster? Or bringing it below default 0.1 will adversely impact the sequencing?

**Reply by mattloose** [reply]
If you reduce this, you will apply the reverse voltage for a shorter period of time. 

This will reduce the amount of time that the pore isn't sequencing for, but it might also reduce the efficiency of the unblocking of reads - i.e a shorter flick may reduce the chance of successfully rejecting a read.

Because we have not tested this we cannot advise on what might happen. We include the parameters because they are things that can be altered.

I hope that helps.

**Comment by jennieli421** [comment]
would there be any adverse effect to setting `throttle` lower?

**Reply by mattloose** [reply]
As answered earlier in this thread, lowering the throttle may result in making too many calls to the API and so reducing your overall performance. As stated earlier you can test this on your system with a playback run to determine the effect.

**Reply by jennieli421** [reply]
Thank you
